src/CMakeLists.txt

```diff
@@ -17,12 +17,20 @@ if (GUI)
     if (WIN32)
         list (APPEND QBT_QT_COMPONENTS WinExtras)
     endif(WIN32)
+    if (APPLE)
+        list (APPEND QBT_GUI_OPTIONAL_LINK_LIBRARIES objc)
+        list (APPEND QBT_QT_COMPONENTS MacExtras)
+    endif (APPLE)
 endif (GUI)
 if (DBUS)
     list (APPEND QBT_QT_COMPONENTS DBus)
 endif (DBUS)
 find_package(Qt5 5.5.1 COMPONENTS ${QBT_QT_COMPONENTS} REQUIRED)
 
+if (GUI AND APPLE)
+    # Fix MOC inability to detect macOS
+    add_definitions(-DQ_OS_MAC)
```

Comment:
> It is saner to define it for all the files, since this flag is defined everywhere but in MOC for all macOS sources, and not defining it may lead to undefined behaviour in other places.

Nevertheless, I would be as specific as possible. Why don't you use [CMAKE_AUTOMOC_MOC_OPTIONS](https://cmake.org/cmake/help/v3.9/variable/CMAKE_AUTOMOC_MOC_OPTIONS.html#variable:CMAKE_AUTOMOC_MOC_OPTIONS) variable?

> If you think a different comment is more descriptive, would you suggest anything? Maybe adding some Qt bug links?

Exactly. Please, add such a comment.